File: java/src/main/java/com/google/api/services/samples/authorizedbuyers/marketplace/v1/buyers/proposals/AddNoteToProposals.java
/*
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.api.services.samples.authorizedbuyers.marketplace.v1.buyers.proposals;

import com.google.api.services.authorizedbuyersmarketplace.v1.AuthorizedBuyersMarketplace;
import com.google.api.services.authorizedbuyersmarketplace.v1.model.AddNoteRequest;
import com.google.api.services.authorizedbuyersmarketplace.v1.model.Note;
import com.google.api.services.authorizedbuyersmarketplace.v1.model.Proposal;
import com.google.api.services.samples.authorizedbuyers.marketplace.Utils;
import java.io.IOException;
import java.security.GeneralSecurityException;
import net.sourceforge.argparse4j.ArgumentParsers;
import net.sourceforge.argparse4j.inf.ArgumentParser;
import net.sourceforge.argparse4j.inf.ArgumentParserException;
import net.sourceforge.argparse4j.inf.Namespace;

/** Adds a note to a given proposal. */
public class AddNoteToProposals {

  public static void execute(AuthorizedBuyersMarketplace marketplaceClient, Namespace parsedArgs) {
    Integer accountId = parsedArgs.getInt("account_id");
    String proposalId = parsedArgs.getString(("proposal_id"));
    String proposalName = String.format("buyers/%d/proposals/%s", accountId, proposalId);
    String noteValue = parsedArgs.getString("note");

    Note note = new Note();
    note.setNote(noteValue);

    AddNoteRequest addNoteRequest = new AddNoteRequest();
    addNoteRequest.setNote(note);

    Proposal proposal = null;
    try {
      proposal =
          marketplaceClient.buyers().proposals().addNote(proposalName, addNoteRequest).execute();
    } catch (IOException ex) {
      System.out.printf("Marketplace API returned error response:%n%s", ex);
      System.exit(1);
    }

    System.out.printf("Adding note to proposal with name \"%s\":%n", proposalName);
    Utils.printProposal(proposal);
  }

  public static void main(String[] args) {
    ArgumentParser parser =
        ArgumentParsers.newFor("AddNoteToProposals")
            .build()
            .defaultHelp(true)
            .description(("Adds a note to a given proposal."));
    parser
        .addArgument("-a", "--account_id")
        .help(
            "The resource ID of the buyers resource under which the proposal is being "
                + "accessed. This will be used to construct the proposal name used as a path "
                + "parameter for the proposals.addNote request.")
        .required(true)
        .type(Integer.class);
    parser
        .addArgument("-p", "--proposal_id")
        .help(
            "The resource ID of the proposals resource that a note is being added to. This "
                + "will be used to construct the name used as a path parameter for the "
                + "proposals.addNote request.")
        .required(true)
        .type(String.class);
    parser
        .addArgument("-n", "--note")
        .help("The note to be added to the proposal.")
        .type(String.class)
        .setDefault("Created note from Java sample.");

    Namespace parsedArgs = null;
    try {
      parsedArgs = parser.parseArgs(args);
    } catch (ArgumentParserException ex) {
      parser.handleError(ex);
      System.exit(1);
    }

    AuthorizedBuyersMarketplace client = null;
    try {
      client = Utils.getMarketplaceClient();
    } catch (IOException ex) {
      System.out.printf("Unable to create Marketplace API service:%n%s", ex);
      System.out.println("Did you specify a valid path to a service account key file?");
      System.exit(1);
    } catch (GeneralSecurityException ex) {
      System.out.printf("Unable to establish secure HttpTransport:%n%s", ex);
      System.exit(1);
    }

    execute(client, parsedArgs);
  }
}

File: php/Examples/V1/Buyers_Proposals/AddNoteToProposals.php
<?php

/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

namespace Google\Ads\AuthorizedBuyers\Marketplace\Examples\V1\Buyers_Proposals;

use Google\Ads\AuthorizedBuyers\Marketplace\ExampleUtil\BaseExample;
use Google\Ads\AuthorizedBuyers\Marketplace\ExampleUtil\Config;
use Google\Service\AuthorizedBuyersMarketplace\AddNoteRequest;
use Google\Service\AuthorizedBuyersMarketplace\Note;

/**
 * This example illustrates how to add a note for the given buyer account and proposal IDs.
 *
 * This note will be visible to the seller and can be used to facilitate the negotiation process.
 */
class AddNoteToProposals extends BaseExample
{
    public function __construct($client)
    {
        $this->service = Config::getGoogleServiceAuthorizedBuyersMarketplace($client);
    }

    /**
     * @see BaseExample::getInputParameters()
     */
    protected function getInputParameters()
    {
        return [
            [
                'name' => 'account_id',
                'display' => 'Account ID',
                'description' =>
                    'The resource ID of the buyers resource under which the proposal was ' .
                    'created. This will be used to construct the name used as a path parameter ' .
                    'for the proposals.addNote request.',
                'required' => true
            ],
            [
                'name' => 'proposal_id',
                'display' => 'Proposal ID',
                'description' =>
                    'The resource ID of the buyers.proposals resource for which the proposal ' .
                    'was created. This will be used to construct the name used as a path ' .
                    'parameter for the proposals.addNote request.',
                'required' => true
            ],
            [
                'name' => 'note',
                'display' => 'Note',
                'description' => 'The note to be added to the proposal.',
                'required' => false,
                'default' => 'Created note from PHP sample.'
            ]
        ];
    }

    /**
     * @see BaseExample::run()
     */
    public function run()
    {
        $values = $this->formValues;

        $name = "buyers/$values[account_id]/proposals/$values[proposal_id]";

        $addNoteRequest = new AddNoteRequest();
        $addNoteRequest->note = Note();
        $addNoteRequest->note->note = $values['note'];

        print "<h2>Adding note to proposal with name '$name':</h2>";
        $result = $this->service->buyers_proposals->addNote($name, $addNoteRequest);
        $this->printResult($result);
    }

    /**
     * @see BaseExample::getName()
     */
    public function getName()
    {
        return 'Add Note to Proposal';
    }
}
